### PR TITLE
feedbackEndpointV1

### PR DESCRIPTION
**Pull request by jonathanwgoodwin**
Simple endpoint to pipe feedback from the UI through to our slack blazar room for response.

### CODE REVIEW

File: BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    return slackWebhookUrl;
+  }
+
+  public String getBaseUrl() {
+    return BASE_URL;
+  }
+
+  public String getBlazarIconLink() {
+    return BLAZAR_ICON_LINK;
+  }
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java]: This method is unused?

File: BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    return room;
+  }
+
+  public String getSlackWebhookUrl() {
+    return slackWebhookUrl;
+  }
+
+  public String getBaseUrl() {
+    return BASE_URL;
+  }
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java]: Unused?

File: BlazarService/src/main/java/com/hubspot/blazar/config/BlazarConfiguration.java [hunk 3]
```diff
@@ -37,6 +37,10 @@
   @JsonProperty("ui")
   private UiConfiguration uiConfiguration = new UiConfiguration();
 
+  @Valid
+  @JsonProperty("slack")
+  private SlackConfiguration slackConfiguration;
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/config/BlazarConfiguration.java]: I think this should be `Optional<SlackConfiguration>` and default to `Optional.absent();`

File: BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package com.hubspot.blazar.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class SlackConfiguration {
+
+  private final String token;
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java]: `@NotNull`
`@NotEmpty`

File: BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  private final String room;
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java]: `@NotNull`
`@NotEmpty`

File: BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  private String slackWebhookUrl;
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java]: `@NotNull`
`@NotEmpty`
`final`

File: BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+  public String getToken() {
+    return token;
+  }
+
+  public String getRoom() {
+    return room;
+  }
+
+  public String getSlackWebhookUrl() {
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/config/SlackConfiguration.java]: Maybe just `getUrl` to match the `@JsonProperty("url")` in the constructor?

File: BlazarService/src/main/java/com/hubspot/blazar/guice/SlackModule.java [hunk 8]
```diff
@@ -0,0 +1,4 @@
+package com.hubspot.blazar.guice;
+
+public class SlackModule {
+}
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/guice/SlackModule.java]: ?

File: BlazarService/src/main/java/com/hubspot/blazar/resources/SlackResource.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+  @Inject
+  public SlackResource(BlazarConfiguration configuration) {
+    this.slackConfiguration = configuration.getSlackConfiguration();
+  }
+
+  @POST
+  @Path("/")
+  public void sendMessage(Feedback feedback) throws IOException {
+    if (!slackConfiguration.isPresent() && slackConfiguration.get().getToken().isEmpty() || slackConfiguration.get().getRoom().isEmpty()) {
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/resources/SlackResource.java]: This conditional seems wrong?

File: BlazarService/src/main/java/com/hubspot/blazar/resources/SlackResource.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+  private DateTimeFormatter dateFormat = DateTimeFormatter.ISO_LOCAL_DATE;
+
+
+  @Inject
+  public SlackResource(BlazarConfiguration configuration) {
+    this.slackConfiguration = configuration.getSlackConfiguration();
+  }
+
+  @POST
+  @Path("/")
```
Comment (jhaber) [inline BlazarService/src/main/java/com/hubspot/blazar/resources/SlackResource.java]: Maybe "/feedback"

File: BlazarBase/src/main/java/com/hubspot/blazar/base/feedback/Feedback.java [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+public class Feedback {
+
+  private final String username;
+  private final String message;
+  private final String page;
+  private final long timestamp;
```
Comment (jhaber) [inline BlazarBase/src/main/java/com/hubspot/blazar/base/feedback/Feedback.java]: I think we can drop `timestamp` from here and just do `System.currentTimeMillis()` in the resource

**Comment by jhaber** [comment]
Left a few more minor things otherwise lgtm

**Comment by jonathanwgoodwin** [comment]
@jhaber Oh makes sense. I can set it up so that:
```
Post to /feedback : feedback sent to all integrations
Post to /feedback/slack: feedback sent to just slack
```

Does that sound good to you?

**Pull request merged**